File: backend/src/main/java/com/example/backend/board/domain/BoardComment.java
package com.example.backend.board.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.springframework.data.relational.core.mapping.Table;
import javax.persistence.*;
import java.sql.Timestamp;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "TEST_DB")
@Entity
public class BoardComment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long commentId;

    @Column(nullable = false)
    private String commentWriter;

    @Column(nullable = false)
    private String commentContent;

    @Column(nullable = false)
    private Timestamp commentDate;

    // FK
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_id")
    @JsonIgnore
    private Board board;
}

File: frontend/frontend/Models/BoardComment.swift
//
//  BoardComment.swift
//  frontend
//
//  Created by 노유수 on 2023/05/21.
//

import Foundation


struct BoardComment: Hashable, Codable{
  let commentId: Int
  let commentWriter: String
  let commentContent: String
  let commentDate: String
}
